node/test/dirtests.ts

```diff
@@ -96,6 +96,51 @@ describe('Dir Operation Tests', function () {
 
         done();
     });
+
+    // which tests
+    it('which() finds executable with scoped permissions', function (done) {
```

Comment:
should this be four tests? one for owner, one for group, one for everyone, one for RW_RW_RW_.

Reply:
or a foreach that runs the same test, but sets the mode differently on each iteration

Reply:
Yeah, this is a good idea. I think I like having separate tests for understandability/descriptive test names rather than a foreach (it can be pulled out into a function though)